### ISSUE TITLE
Documentation about the default culture of new threads is outdated

### ISSUE THREAD

**Issue by cremor** [dotnet/dotnet-api-docs#3584]
On the API documentation page of [CultureInfo.DefaultThreadCurrentCulture](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentculture?view=netframework-4.8) there is a sample that outputs the culture which is used by new threads by default. I've ran this sample on my machine and couldn't get the documented output when I used a recent target framework. Specifically, the documented output of the first sample and this sentence below it is wrong:

> However, because the worker thread's culture is derived from the current Windows system culture rather than the application's current culture, the work thread displays its currency values using the formatting conventions of the English (United States) culture.

Here are my test results:

- Runtime 4.8, target framework 4.5.2 (or lower): Documentation is correct
- Runtime 4.8, target framework 4.6-4.8: Documentation is wrong
- .NET Core 2.0-3.0: Documentation is wrong

With "Documentation is wrong" I always mean that the started worker thread **does** use the same culture as the main thread. For example, this is the output I get:

```
Current Culture:    fr-FR
Current UI Culture: fr-FR
Random Values:      0,08 €          0,34 €          0,99 €

Current Culture:    fr-FR
Current UI Culture: fr-FR
Random Values:      0,88 €          0,21 €          0,75 €
```

There are quite a few pages that claim that the default culture of new threads is the Windows culture, so all those need to be updated:
- [CultureInfo](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo?view=netframework-4.8)
- [CultureInfo.DefaultThreadCurrentCulture](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentculture?view=netframework-4.8)
- [CultureInfo.DefaultThreadCurrentUICulture](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.defaultthreadcurrentuiculture?view=netframework-4.8)
- [CultureInfo.CurrentCulture](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentculture?view=netframework-4.8)
- [CultureInfo.CurrentUICulture](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentuiculture?view=netframework-4.8)
- [Thread](https://docs.microsoft.com/en-us/dotnet/api/system.threading.thread?view=netframework-4.8)
(There are most likely even more.)

Note: Some of those pages already mention that task-based asynchronous operations flow the culture. But as my tests show, this is also true for "normal" threads.

**Comment by mairaw** [comment]
Thanks for your feedback @cremor. @krwq @tarekgh can you please check this feedback?

**Comment by tarekgh** [comment]
I agree with @cremor the doc may need some clarifications and updates. Here are some comments which can help to clarify: 

- It is not recommended to use Thread.CurrentCulture/CurrentUICulture because it causes some confusion as these properties will set the current cultures to the current threads and not to the thread object used in the call. Users should treat these properties as obsolete and always should use CultureInfo.CurrentCulture/CurrentUICulture. We need to mention that in the doc but still this will not be enough to address the issues raised by @cremor here. 
- In the full framework, starting from version 4.6 we had dome some changes to have the current cultures flow with the current thread execution context. The [doc](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/appcontextswitchoverrides-element#remarks) mentioning the following: 

| Switch name | Description | Introduced|
| --- | --- |--- |
| Switch.System.Globalization.NoAsyncCurrentCulture | Controls whether asynchronous operations do not flow from the calling thread's context. For more information, see CurrentCulture and CurrentUICulture flow across tasks. | .NET Framework 4.6 |

- There is a way to disable the new behavior on the full framework by using the config switch: 
```XML
<configuration>
   <runtime>
      <AppContextSwitchOverrides value="Switch.System.Globalization.NoAsyncCurrentCulture=true" />
   </runtime>
</configuration>
``` 
- The new current culture's behavior which flows the cultures with the execution context to the created threads is the default behavior on .Net Core and cannot be disabled by any config switch.
- You cannot stop flowing the current execution context to the new threads by calling `ExecutionContext.SuppressFlow()`. So in the first code example in the doc which current producing a wrong result can be fixed by just adding the call to `ExecutionContext.SuppressFlow()` to the code before creating the new thread.
- I agree with updating all references to Windows culture to change to something to OS culture instead. 

Thanks, @cremor for reporting this issue.

@mairaw please let me know if you need more information to fix the docs.

**Comment by cremor** [comment]
> You cannot stop flowing the current execution context to the new threads by calling `ExecutionContext.SuppressFlow()`. So the first code example in the doc which currently producing a wrong result can be fixed by just adding the call to `ExecutionContext.SuppressFlow()` to the code before creating the new thread.

@tarekgh  Something is not right here. Did you mean to write "You **can** stop flowing..."?
If yes, is it really good to fix the example in that way? Shouldn't it better be updated to show the default behavior of .NET Core and current versions of .NET Framework?

**Comment by tarekgh** [comment]
> Something is not right here. Did you mean to write "You can stop flowing..."?

Yes, I meant You can stop. I'll fix that. thanks for spotting it. 

> Shouldn't it better be updated to show the default behavior of .NET Core and current versions of .NET Framework?

I am not sure if you are referring to the doc or the code sample here. anyway, I think we need to update both. but the code sample will just need to use `ExecutionContext.SuppressFlow()`. please let me know if you mean something else.

**Comment by cremor** [comment]
I meant the code sample on the documentation page of `CultureInfo.DefaultThreadCurrentCulture`. If I understand you correctly, you suggest to add `ExecutionContext.SuppressFlow()` in there so that it once again doesn't flow the culture and produces the documented output.
My alternative suggestion would be to update the sample in a way that shows that the culture does flow by default. (Maybe that means just updating the expected output.) Then it would still show the default behavior of the .NET Framework (at least of current versions), as indicated by the first sentence in the "examples" section.

**Comment by tarekgh** [comment]
> My alternative suggestion would be to update the sample in a way that shows that the culture does flow by default. (Maybe that means just updating the expected output.) Then it would still show the default behavior of the .NET Framework (at least of current versions), as indicated by the first sentence in the "examples" section.

This is a good suggestion but I believe the code sample should be in the CultureInfo.CurrentCulture doc page instead. There we can mention more details as needed.
